src/test/java/uk/gov/hmcts/ccd/domain/types/ApprovalStatusValidatorTest.java

```diff
@@ -0,0 +1,107 @@
+    private CaseFieldDefinition caseFieldDefinition;
+
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.initMocks(this);
+
+        when(definitionRepository.getBaseTypes()).thenReturn(Collections.emptyList());
+        BaseType.setCaseDefinitionRepository(definitionRepository);
+        BaseType.initialise();
+
+        when(numberBaseType.getType()).thenReturn(NumberValidator.TYPE_ID);
+        BaseType.register(numberBaseType);
+
+        validator = new ApprovalStatusValidator();
+
+        caseFieldDefinition = caseField(FIELD_ID).build();
+    }
+
+    private CaseFieldDefinitionBuilder caseField(String fieldId) {
+        return new CaseFieldDefinitionBuilder(fieldId).withType(ApprovalStatusValidator.TYPE_ID);
+    }
+
+    @Test
+    void shouldFailOnInvalidApprovalStatus() {
+        final JsonNode data = NODE_FACTORY.textNode("5");
+        List<ValidationResult> validationResults =
+            this.validator.validate("ApprovalStatus", data, caseFieldDefinition);
+        assertNotNull(validationResults);
+        assertEquals(1, validationResults.size());
+        assertEquals("Invalid Approval Status Value, Valid values are 0,1 and 2. "
+                + "0 = ‘Not considered’, 1 = ‘Approved’, 2 = ‘Rejected’",
+            validationResults.get(0).getErrorMessage());
+    }
+
+    @Test
+    void shouldValidateOnValidApprovalStatus() {
+        final JsonNode data = NODE_FACTORY.textNode("0");
+        List<ValidationResult> validationResults =
+            this.validator.validate("ApprovalStatus", data, caseFieldDefinition);
+        assertNotNull(validationResults);
+        assertEquals(0, validationResults.size());
+    }
+
+    @Test
+    void shouldReturnEmptyErrorListValidateForOtherFieldId() {
+        caseFieldDefinition = caseField(ORGANISATION_TO_ADD).build();
+        final JsonNode data = NODE_FACTORY.textNode("0");
+        List<ValidationResult> validationResults =
+            this.validator.validate("OrganisationToAdd", data, caseFieldDefinition);
+        assertNotNull(validationResults);
+        assertEquals(0, validationResults.size());
+    }
+
+    @Test
+    void shouldReturnEmptyErrorListValidateForEmptyData() {
+        final JsonNode data = NODE_FACTORY.textNode("");
+        List<ValidationResult> validationResults =
+            this.validator.validate("OrganisationToAdd", data, caseFieldDefinition);
+        assertNotNull(validationResults);
+        assertEquals(0, validationResults.size());
+    }
+
+    @Test
+    void shouldReturnEmptyErrorListValidateForNullData() {
+        final JsonNode data = null;
+        List<ValidationResult> validationResults =
+            this.validator.validate("OrganisationToAdd", data, caseFieldDefinition);
+        assertNotNull(validationResults);
+        assertEquals(0, validationResults.size());
+    }
+
+    @Test
+    void getType() {
+        assertEquals(validator.getType(), BaseType.get("Number"), "Type is incorrect");
+    }
+}
```

Comment:
There is a class that test all validations together based on definitions.  CaseDataValidatorTest . May be you can add your new Type and its validator there.  It is just a recommendation . Not mandatory.

Reply:
we are not checking data here.. this is check checking the method return type, which is static in this case